lib/samples/snap_geometry_edits/snap_geometry_edits.dart

```diff
@@ -574,9 +585,36 @@ class _SnapGeometryEditsState extends State<SnapGeometryEdits>
     return Column(
       crossAxisAlignment: CrossAxisAlignment.start,
       children: [
-        Text(
-          label,
-          style: Theme.of(context).textTheme.titleMedium,
+        Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            Text(
+              label,
+              style: Theme.of(context).textTheme.titleMedium,
+            ),
+            Row(
+              children: [
+                const Text('Enable all'),
+                // A checkbox to enable all source settings in the category.
+                Checkbox(
+                  value: allSourceSettings.every(
+                    (snapSourceSettings) => snapSourceSettings.isEnabled,
+                  ),
+                  onChanged: (allEnabled) {
+                    if (allEnabled != null) {
+                      allSourceSettings
+                          .map(
+                            (snapSourceSettings) => setState(
+                              () => snapSourceSettings.isEnabled = allEnabled,
+                            ),
+                          )
+                          .toList();
```

Comment:
I don't like using `toList()` for its side-effect (to consume the iterable) while having no use for the returned List. But `forEach` causes a dumb warning and `for (final x in y)` can't be chained. So I dunno, I guess `.toList()` is the best option and is pretty idiomatic in that all Flutter developers end up doing it this way. Unless we want to be really picky and use a `for` loop in this scenario?

Reply:
It is a bit sort of not right isn't it. The forEach thing is really annoying. 
I'm leaning towards leaving it - I think it still feels cleaner this way. Let's see if this arises again and at a non-release time can potentially discuss more.